src/core/Form/SearchInput/SearchInput.test.tsx

```diff
@@ -199,6 +199,45 @@ describe('props', () => {
       }
     });
   });
+  describe('debounce', () => {
+    it('delays the running of onChange by the given time', () => {
+      jest.useFakeTimers();
+      const mockOnChange = jest.fn();
+      const searchInput = (
+        <SearchInput
+          labelText="Debounced search"
+          clearButtonLabel="clear"
+          searchButtonLabel="search"
+          debounce={1000}
+          onChange={mockOnChange}
+        />
+      );
+      const { getByRole } = render(searchInput);
+
+      const inputElement = getByRole('searchbox') as HTMLInputElement;
+      fireEvent.change(inputElement, { target: { value: 'new value' } });
+      expect(mockOnChange).not.toBeCalled();
+      jest.advanceTimersByTime(1000);
```

Comment:
_Just a small idea maybe not reason to change._
As the test name implies we want to make sure that `onChange` is delayed by a given time, we could check if that is really the case. As now we are advancing the time only once and straight to wanted.
But will it call `onChange` already with lower delay?

It could be replaced like this:
```js
      jest.advanceTimersByTime(500);
      expect(mockOnChange).not.toBeCalled();
      jest.advanceTimersByTime(500);
```

Then we know that by delaying only 500, it was not run yet.

Reply:
Don't know the inner workings of the advanceTimersByTime, but doing the above might be problematic since we are doing an expect in between two timers and that might lead into situation where expect is run actually later than 500ms after the change event. This might depend on the environment as well. If advanceTimers is smart, this might work well too.

Reply:
It would be logical that the fake timers are only advanced when called to do so. That'd support testing with the extra step in between. Seems to work that way, so I'll add it.

Reply:
The issue I was expecting was about setTimeOut which we cannot fake and should resolve on its own regardless of fake timers. As further testing proved, in this case there is no such issue and the code is synchronous so this can be left as is.